libr/util/idpool.c

```diff
@@ -163,6 +163,39 @@ R_API void* r_id_storage_get(RIDStorage* storage, ut32 id) {
 	return storage->data[id];
 }
 
+// usually these shall not be used
+// id-storage doesn't provide sorting things in specific order
```

Comment:
Is it really true? I think there is indeed sorting provided by id, afaik, as things are stored by id.

Reply:
idstorage was created during siol as a replacement for sdb. I felt like using sdb to store int-values from idpool was just unnecessary perf consuming (sprintf), for no reason. idstorage is a thing that you pass a pointer that you want to store and it, and it grabs an id from idpool and uses that as a key, which it passes back to the caller. Cool thing about it, is that it always will use the same "pattern" of keys/ids, whenever you run r2 and do the same commands. This makes creating r2-scripts much easier, as you can just hardcode some values, bc you know, it will always with these values.
Ofc, you could do sort operations on idstorage, but I felt like this could weaken the scripting reliability of io. Therefor there is no "r_id_storage_sort". if one wants to sort things in an idstorage, i suggest using oidstorage, which is basically an idstorage with a permutation array, on which one can do sorting operations.

I hope this explains, why I don't see these ids as a sorting

Reply:
it's a bit fuzzy but... ok :D